demos/mqtt/mqtt_demo_plaintext/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+set( DEMO_NAME "mqtt_demo_plaintext" )
+# Demo target.
+add_executable(${DEMO_NAME})
+
+target_sources(
+    ${DEMO_NAME}
+    PRIVATE
+        "${DEMO_NAME}.c"
+)
+
+target_link_libraries(
+    ${DEMO_NAME}
+    PRIVATE
+        mqtt   
+)
+
+target_include_directories(
+    mqtt
```

Comment:
I think `mqtt` should be fine to be considered as a public dependency for the demo